lib/auth/auth_with_roles_test.go

```diff
@@ -8449,28 +8448,50 @@ func TestCloudDefaultPasswordless(t *testing.T) {
 				},
 			})
 			require.NoError(t, err)
-			_, err = srv.AuthServer.UpsertAuthPreference(ctx, authPreference)
+			_, err = srv.Auth().UpsertAuthPreference(ctx, authPreference)
 			require.NoError(t, err)
 
 			// create preexisting users
 			for i := 0; i < tc.qtyPreexistingUsers; i += 1 {
-				_, _, err = CreateUserAndRole(srv.AuthServer, fmt.Sprintf("testuser-%d", i), nil /* allowedLogins */, nil /* allowRules */)
+				_, _, err = CreateUserAndRole(srv.Auth(), fmt.Sprintf("testuser-%d", i), nil /* allowedLogins */, nil /* allowRules */)
 				require.NoError(t, err)
 			}
 
-			// add the user that might have the auth method changed to passwordless
-			username := "passwordless-user"
-			_, _, err = CreateUserAndRole(srv.AuthServer, username, nil /* allowedLogins */, nil /* allowRules */)
+			// create the user that might have the auth method changed to passwordless
+			u, err := createUserWithSecondFactors(srv)
+			require.NoError(t, err)
+
+			// add permission to edit users, necessary to change auth method to passwordless
+			user, err := types.NewUser(u.username)
+			require.NoError(t, err)
+			role := services.RoleForUser(user)
+			role.SetRules(types.Allow, []types.Rule{
+				types.NewRule(types.KindUser, services.RW()),
+			})
+			upsertedRole, err := srv.Auth().UpsertRole(ctx, role)
+			require.NoError(t, err)
+			user.AddRole(upsertedRole.GetName())
+			srv.Auth().UpdateUser(ctx, &types.UserV2{})
+			_, err = srv.Auth().UpsertUser(ctx, user)
+			require.NoError(t, err)
+
+			client, err := srv.NewClient(TestUser(user.GetName()))
 			require.NoError(t, err)
 
+			client.SetMFAPromptConstructor(func(po ...mfa.PromptOpt) mfa.Prompt {
+				return mfa.PromptFunc(func(ctx context.Context, chal *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error) {
+					return u.webDev.SolveAuthn(chal)
+				})
+			})
```

Comment:
Is this needed if we are solving the challenges explicitly below?

Reply:
No, thank you!